### ISSUE TITLE
Add role to check certificate secret renewal

### ISSUE THREAD

**Issue by JonahLuckett** [ibm-mas/ansible-devops#327]
The role will compare secrets both before and after running the renewal command `cmctl renew` to renew internal certificates and check that this has changed the status from `ready` to `renewing`.

Checking that the new secret is different from the old one will test that cert-manager has done its job

See https://cert-manager.io/docs/usage/cmctl/#renew

**Comment by JonahLuckett** [comment]
@durera if I can get assigned to this issue that'd be perfect

**Comment by JonahLuckett** [comment]
No worries - it will run the renewal command and check that a different secret has been created so the renewal will actually take place. The initial plan is to run this against just one service (internalApi) and verify after renewal that the new certificate is what is being presented and check the health check endpoints. So the final test will be installing mas then renewing certificates as that way the other tests will be helping us validate certs. Does that make sense?

**Comment by whitfiea** [comment]
@JonahLuckett yes I think that makes sense. So this is a role that you would want to only run once as otherwise it would constantly renew the certificates on each run? That is unless we have some condition that means we only renew when we need too i.e. that the CA in the issuer has changed so we need to renew all the certificates not signed by this update issuer?

All other roles are idempotent so running the db2u role 100 times for example against a cluster won't cause 100 installs of db2u but just one install (from the first run) and all other runs of the role just ensures that the state of what we want db2u to be is correct.

**Comment by andrercm** [comment]
@whitfiea @JonahLuckett  any news or plans to work on this issue?
I'm working on triaging all the list of issues in our backlog so we can prioritize them accordingly.

**Comment by JonahLuckett** [comment]
@andrercm Thanks for the bump - will close this issue as the work for this problem was solved - I closed the [internal issue](https://github.ibm.com/wiotp/tracker/issues/9081) at the end of last year
